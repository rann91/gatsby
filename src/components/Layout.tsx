import React from 'react'
import { PageProps } from 'gatsby'

const Layout = ({ children }: PageProps) => (
  <>
    <main>{children}</main>
  </>
)

export default Layout
