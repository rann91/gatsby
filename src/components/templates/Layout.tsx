/* @jsx jsx */
import { ApolloProvider } from '@apollo/client'
import { PageProps } from 'gatsby'
import { Flex, jsx } from 'theme-ui'
import client from '../../graphql/client'
import Footer from '../organisms/Footer'
import Header from '../organisms/Header'

import 'typeface-montserrat/index.css'

const Layout = ({ children }: PageProps) => (
  <ApolloProvider client={client}>
    <Flex
      sx={{
        flexDirection: 'column',
        minHeight: '100vh'
      }}>
      <Header />
      <main sx={{ flex: 1 }}>{children}</main>
      <Footer />
    </Flex>
  </ApolloProvider>
)

export default Layout
