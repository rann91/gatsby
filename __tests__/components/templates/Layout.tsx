import { render } from '@testing-library/react'
import { useStaticQuery } from 'gatsby'
import React from 'react'
import Layout from '../../../src/components/templates/Layout'
import footerData from '../../fixtures/footer.json'
import navigationData from '../../fixtures/navigation.json'
import siteMetadata from '../../fixtures/site-metadata.json'

beforeEach(() => {
  ;(useStaticQuery as jest.Mock)
    .mockImplementationOnce(() => siteMetadata)
    .mockImplementationOnce(() => navigationData)
    .mockImplementationOnce(() => footerData)
})

describe('Layout', () => {
  it('renders correctly', () => {
    // @ts-ignore
    const { container } = render(<Layout />)

    expect(container).toMatchSnapshot()
  })
})
