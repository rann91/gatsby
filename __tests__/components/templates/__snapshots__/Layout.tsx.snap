// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div>
  <div
    class="css-13q1v4g-Layout"
  >
    <header
      class="css-14nc0uu-Header"
      data-testid="header"
    >
      <div
        class="css-zk5be6-Header"
      >
        <div
          class="css-i1z8pl"
        >
          <a
            class="css-13dgcig-Branding"
            data-testid="branding"
            href="/"
          >
            <div
              class="css-18ov0hz-Branding"
            >
              <div
                class="css-fhum7v-Branding"
              >
                <img
                  alt="Branding Logo"
                  class="css-9taffg"
                  src="test-file-stub"
                />
              </div>
              <div
                class="css-vurnku"
              >
                Gatsby
              </div>
            </div>
          </a>
        </div>
        <div
          class="css-i1z8pl"
        >
          <div
            class="css-1l0o9ue-PureNavigation"
          >
            <button
              aria-label="Open Menu"
              class="css-wwcqj1"
              title="Menu"
            >
              <svg
                class="css-ipc245"
                fill="currentcolor"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                />
              </svg>
            </button>
          </div>
          <nav
            class="css-4swsle-PureNavigation"
          >
            <ul
              class="css-13786ft-PureNavigation"
            >
              <li
                class="css-1w5hvub"
              >
                <a
                  class="css-1ys58o5"
                  data-testid="gatsby-link"
                  href="/"
                  target="_self"
                >
                  Home
                </a>
              </li>
              <li
                class="css-1w5hvub"
              >
                <a
                  class="css-1ys58o5"
                  data-testid="gatsby-link"
                  href="/about"
                  target="_self"
                >
                  About
                </a>
              </li>
              <li
                class="css-1w5hvub"
              >
                <a
                  class="css-1ys58o5"
                  data-testid="gatsby-link"
                  href="/work"
                  target="_self"
                >
                  Work
                </a>
              </li>
              <li
                class="css-1w5hvub"
              >
                <a
                  class="css-1ys58o5"
                  data-testid="gatsby-link"
                  href="/team"
                  target="_self"
                >
                  Team
                </a>
              </li>
              <li
                class="css-1w5hvub"
              >
                <a
                  class="css-1ys58o5"
                  data-testid="gatsby-link"
                  href="/blog"
                  target="_self"
                >
                  Blog
                </a>
              </li>
              <li
                class="css-1w5hvub"
              >
                <a
                  class="css-1ys58o5"
                  data-testid="gatsby-link"
                  href="/contact"
                  target="_self"
                >
                  Contact
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="css-1g17dmu-Layout"
    />
    <footer>
      <section
        class="css-fvc22l-Cta"
        data-testid="section"
      >
        <div
          class="css-1yrmf69"
        >
          <div
            class="css-i1z8pl"
          >
            <div
              class="css-ycxng9-TextContainer"
              data-testid="text-container"
            >
              <h2
                class="css-1wf7gym"
              >
                Grow your business
              </h2>
              <p
                class="css-w9nyfe-Cta"
              >
                Today is the day to build the business of your dreams. Share your mission with the world â€” and blow your customers away.
              </p>
              <a
                class="css-1s79t5o-Cta"
                data-testid="gatsby-link"
                href="/about"
                target="_self"
              >
                Learn more
              </a>
            </div>
          </div>
        </div>
      </section>
      <div
        class="css-1ca5803"
        data-testid="bottom"
      >
        <div
          class="css-1yrmf69"
        >
          <p
            class="css-10im984-Bottom"
          >
            Made with
             
            <a
              class="css-vurnku"
              href="https://www.gatsbyjs.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              Gatsby
            </a>
             
            and
             
            <a
              class="css-vurnku"
              href="https://strapi.io"
              rel="noopener noreferrer"
              target="_blank"
            >
              Strapi
            </a>
            . Design by
             
            <a
              class="css-vurnku"
              href="https://webflow.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              Webflow
            </a>
            .
          </p>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
